### PR TITLE
Integration test

### LINKED ISSUE #1056: Integration test
We should have a test that install Got with only production dependencies `npm i --production` and then try to require it and do a request, just to ensure nothing broke in production. For example, that we didn't forget a dependency (https://github.com/sindresorhus/got/issues/1055).

### PR DESCRIPTION
**Pull request by Giotino**
This is just a draft about @sindresorhus proposal (#1056).

Actually this is really simple, after all the tests it removes the dev dependencies and make a request to https://google.com, if it fails (or if something's wrong with the dependencies) it exit with an exit code != 0 resulting in a Travis CI failure.

Working: https://travis-ci.org/github/Giotino/got/builds/687760094
Non working: https://travis-ci.org/github/Giotino/got/builds/687760585

Fixes #1056

### CODE REVIEW

File: integration-test/example.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const got = require('../dist/source');
+
+(async () => {
+	try {
+		await got('https://google.com');
```
Comment (szmarczak) [inline integration-test/example.js:5]: Can you create a local server instead? We should avoid querying 3rd party servers.

Comment (szmarczak) [reply integration-test/example.js:5]: Actually for an integration test... I don't know.

Comment (Giotino) [reply integration-test/example.js:5]: I think these tests should be as close as possible to real usages.

File: package.json [hunk 2]
```diff
@@ -10,10 +10,12 @@
...
 	},
 	"scripts": {
+		"build": "del-cli dist && tsc",
 		"test": "xo && tsc --noEmit && nyc --reporter=html --reporter=text ava",
+		"coverage": "nyc report --reporter=text-lcov | coveralls",
+		"prepare": "npm run build",
 		"release": "np",
-		"build": "del-cli dist && tsc",
-		"prepare": "npm run build"
+		"clean": "rimraf dist coverage .nyc_output"
```
Comment (sindresorhus) [inline package.json:18]:
```suggestion
		"clean": "del-cli dist coverage .nyc_output"
```

File: package.json [hunk 3]
```diff
@@ -10,10 +10,12 @@
 		"node": ">=10.19.0"
 	},
 	"scripts": {
+		"build": "del-cli dist && tsc",
 		"test": "xo && tsc --noEmit && nyc --reporter=html --reporter=text ava",
+		"coverage": "nyc report --reporter=text-lcov | coveralls",
```
Comment (sindresorhus) [inline package.json:15]: I would prefer to keep this in the Travis file.

File: integration-test/example.js (hunk 1 above)
Comment (szmarczak) [reply integration-test/example.js:6]: Can we use 3rd-party packages for this? They use Got more extensively I think.

Comment (szmarczak) [reply integration-test/example.js:6]: Like https://github.com/yarnpkg/berry or https://github.com/renovatebot/renovate or https://github.com/MONEI/Shopify-api-node/ or (it goes on and on).

**Comment by szmarczak** [comment]
IMO this is a good start and we can add more complicated integration tests later.

**Comment by sindresorhus** [comment]
I think the integration test would be a good opportunity to use GitHub Actions.

**Comment by sindresorhus** [comment]
@Giotino Anything blocking you to finish this or are you just busy?

**Comment by Giotino** [comment]
> @Giotino Anything blocking you to finish this or are you just busy?

Yeah, I was busy and since this is not critical to Got I delayed it a little bit.

> I think the integration test would be a good opportunity to use GitHub Actions.

On the GitHub action side, I think we should use the same workflow as TravisCI (without coveralls).
I'm going to add the Github workflow file to this PR.

**Comment by sindresorhus** [comment]
Ok. No worries. 👍🏻

**Comment by szmarczak** [comment]
@Giotino Friendly ping :)

**Comment by sindresorhus** [comment]
> I think the integration test would be a good opportunity to use GitHub Actions.

This was done in https://github.com/sindresorhus/got/commit/35f5822c2029b0075c381663d27aa39608dc0a4d

**Pull request closed without merging**